frontend/src/board/pgn/boardTools/underboard/settings/GameSettings.tsx

```diff
@@ -43,32 +43,26 @@ interface GameSettingsProps {
 const GameSettings: React.FC<GameSettingsProps> = ({ game, onSaveGame }) => {
     const isFreeTier = useFreeTier();
     const [visibility, setVisibility] = useState(
-        game.unlisted ?? true ? 'unlisted' : 'published',
+        game.unlisted ?? false ? 'unlisted' : 'published',
     );
     const [orientation, setOrientation] = useState<BoardOrientation>(
         game.orientation ?? 'white',
     );
-    const [dirty, setDirty] = useState(false);
     const [headers, setHeaders] = useState<PgnHeaders>(game.headers);
 
     const headersChanged = Object.entries(game.headers).some(
         ([name, value]) => value !== headers[name],
     );
 
-    const navigate = useNavigate();
-
-    const onChangeOrientation = (newOrientation: BoardOrientation) => {
-        setDirty(true);
-        setOrientation(newOrientation);
-    };
+    const unlisted = visibility === 'unlisted';
+    const dirty =
+        headersChanged ||
+        orientation !== game.orientation ||
+        (game.unlisted ?? false) !== unlisted;
 
-    const onChangeVisibility = (visibility: string) => {
-        setDirty(true);
-        setVisibility(visibility);
-    };
+    const navigate = useNavigate();
```

Comment:
Can you move this up? I like to keep all the hooks together at the top. It helps prevent accidentally returning early before a hook call

Reply:
Yes, I agree about that best practice. This was a mistake. There exist lint rules to enforce this, by the way.